### PR TITLE
Issue4

### LINKED ISSUE #4: Image Index
#### As a user I want the homepage to display all saved images.

__Story__:

Now that we are saving images for our users, we want them to be able to view
the list of images in the system. You might be thinking, "what if my users only
want to view images of cats", or "what if my users only want to see the images
they uploaded". Those are great questions. Maybe you will have the opportunity
to add those features in a future story. For now, we will keep it simple and
quickly deliver a little more value to our users.

__Acceptance criteria__:
- [ ] New images that are added show up on the homepage.
- [ ] These images are persisted if the browser is closed or even if the
  server is restarted.
- [ ] Images are not displayed wider than 400px.
- [ ] Newest images appear first.

__Dependencies__:
- Save Image Link

### PR DESCRIPTION
**Pull request by aman-singh**
Closes #4 - Ability to add tags for an image and display these tags on show page and home page

### CODE REVIEW

**Comment by aman-singh** [comment]
[Issue 4: Image Tags](https://trello.com/c/CBs2OGIt/791-issue-4-image-tags)

File: app/views/application/home.html.erb [hunk 1]
```diff
@@ -5,5 +5,6 @@
 <% @images.reverse.each do |image| %>
   <p>Create Date: <%= image.created_at %>></p>
   <%= image_tag image.url, class: 'index-img' %>
+  <p>Tags: <%= image.tag_list %></p>
```
Comment (sthust) [inline app/views/application/home.html.erb:8]: can we add similar the first exercise an id for the tags field so that we can later in the test correctly identify that the content of the tags display in the right place on the page?

File: app/views/images/show.html.erb [hunk 2]
```diff
@@ -5,4 +5,5 @@
 <% end %>
 <h1>Aman's image sharer app</h1>
 <h3>Here's the image you uploaded</h3>
+<h5>Tags: <%= @image.tag_list %></h5>
```
Comment (sthust) [inline app/views/images/show.html.erb:8]: similar. let's add and id to correctly identify in the tests that the right content is displayed on the page in the right location.

File: test/models/image_test.rb [hunk 3]
```diff
@@ -10,4 +10,12 @@ class ImageTest < ActiveSupport::TestCase
...
   end
+
+  test 'should be able to add tags' do
+    image = Image.new url: 'https://cdn.pixabay.com/photo/2015/04/23/22/00/tree-736885_960_720.jpg',
+                      tag_list: 'tag1, tag2'
+    assert image.valid?
+    assert_equal 'tag1', image.tag_list[0]
+    assert_equal 'tag2', image.tag_list[1]
+  end
 end
```
Comment (sthust) [inline test/models/image_test.rb:21]: can we also add a test that the tag list can be empty, don't specify a tag when creating a new image and assert that you get an empty array?

Comment (aman-singh) [reply test/models/image_test.rb:21]: Added

File: app/views/images/show.html.erb (hunk 2 above)
Comment (aman-singh) [reply app/views/images/show.html.erb:8]: There's only 1 instance of Tags on this page. Adding an id anyways.

File: app/views/application/home.html.erb (hunk 1 above)
Comment (aman-singh) [reply app/views/application/home.html.erb]: done
